libcloud/container/drivers/lxd.py

```diff
@@ -662,10 +730,17 @@ def destroy_container(self, container, ex_timeout=default_time_out):
 
         # Return: background operation or standard error
         req = '/%s/containers/%s' % (self.version, container.name)
-        response = self.connection.request(req, method='DELETE')
 
-        response_dict = response.parse_body()
-        assert_response(response_dict=response_dict, status_code=100)
+        try:
+            response = self.connection.request(req, method='DELETE')
+
+            response_dict = response.parse_body()
+            assert_response(response_dict=response_dict, status_code=100)
+        except BaseHTTPError as e:
+            # handle the case where the container is running
+            message_list = e.message.split(",")
```

Comment:
I see this logic duplicated across multiple methods so it would perhaps be better to refactor it into some common method - e.g. ``_get_error_message_for_exception`` or similar.

And some tests for this method would also be good.

Reply:
EDIT: Or perhaps even, ``_get_lxd_ap_exception_for_error`` and returning ``LXDAPIException`` class instance.